src/nemo_safe_synthesizer/pii_replacer/core.py

```
@@ -0,0 +1,2898 @@
+_TITLE_RE = re.compile(r"^(dr|mr|mrs|ms|miss|prof|sir|madam|mx)\.?\s+", re.IGNORECASE)
+
+
+def split_title(value: str):
+    m = _TITLE_RE.match(value.strip())
+    if m:
+        return value[: m.end()].strip(), value[m.end() :].strip()
+    return None, value.strip()
+
+
+def _stable_hash(s: str) -> int:
+    return int(hashlib.md5(s.encode("utf-8")).hexdigest()[:8], 16)
```

Ruff (0.16.0)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)